viz_scripts/generic_metrics.ipynb

```diff
@@ -101,192 +102,493 @@
+    "    if not expanded_ct_sensed.empty:\n",
+    "        df_confirm_dist_land = process_distance_data(expanded_ct[expanded_ct['Mode_confirm'] != \"Airplane\"], 'Mode_confirm', \"Labeled by user\\n (Trip distance - Land)\")\n",
+    "        if not df_confirm_dist_land.empty:\n",
+    "            bar_count += 1\n",
+    "            all_data_frames.append(df_confirm_dist_land)\n",
+    "        else:\n",
+    "            print(\"df_confirm_dist_land is empty.\")\n",
+    "    else:\n",
+    "        print(\"expanded_ct is empty.\")\n",
     "\n",
-    "    labels_d10 = expanded_ct.loc[(expanded_ct['distance'] <= cutoff)].Mode_confirm.value_counts(dropna=True).keys().tolist()\n",
-    "    values_d10 = expanded_ct.loc[(expanded_ct['distance'] <= cutoff)].Mode_confirm.value_counts(dropna=True).tolist()\n",
-    "    d10_quality_text = scaffolding.get_quality_text(expanded_ct, expanded_ct[expanded_ct['distance'] <= cutoff], \"< \" + dist_threshold + \" \" + label_units_lower, include_test_users)\n",
-    "    plot_title = plot_title_no_quality+\"\\n\"+d10_quality_text\n",
-    "    pie_chart_mode(plot_title,labels_d10,values_d10,file_name)\n",
-    "    alt_text = store_alt_text_pie(pd.DataFrame(values_d10, labels_d10), file_name, plot_title)\n",
-    "    print(expanded_ct.loc[(expanded_ct['distance'] <= cutoff)].Mode_confirm.value_counts(dropna=True))\n",
+    "    if not expanded_ct_sensed.empty:\n",
+    "        df_sensed_dist_land = process_distance_data(expanded_ct_sensed[expanded_ct_sensed['primary_mode'] != \"AIR_OR_HSR\"], 'primary_mode', \"Sensed by OpenPATH\\n (Trip distance - Land)\")\n",
+    "        if not df_sensed_dist_land.empty:\n",
+    "            bar_count += 1\n",
+    "            all_data_frames.append(df_sensed_dist_land)\n",
+    "        else:\n",
+    "            print(\"df_sensed_dist_land is empty.\")\n",
+    "    else:\n",
+    "        print(\"expanded_ct_sensed is empty.\")\n",
     "\n",
+    "    result_df = merge_dataframes(all_data_frames)\n",
+    "    stacked_bar_chart_generic(plot_title, result_df, file_name, bar_count)\n",
+    "    alt_text, alt_html = store_alt_text_stacked_bar_chart(result_df[result_df['Count'] > 0], file_name, plot_title)\n",
     "except:\n",
-    "    d10_df = expanded_ct.query(\"distance <= \" + str(cutoff)) if \"distance\" in expanded_ct.columns else expanded_ct\n",
-    "    debug_df.loc[\"Trips_less_than_80th_pct\"] = scaffolding.trip_label_count(\"Mode_confirm\", d10_df)\n",
-    "    generate_missing_plot(plot_title_no_quality,debug_df,file_name)\n",
-    "    alt_text = store_alt_text_missing(debug_df, file_name, plot_title_no_quality)"
+    "    generate_missing_plot(plot_title_no_quality, merged_debug_df, file_name)\n",
+    "    alt_text = store_alt_text_missing(merged_debug_df, file_name, plot_title_no_quality)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "dominant-company",
+   "id": "7aa03722",
```

Comment:
nit: why has the ID format changed? Are you using the same version of the ipython notebook?

Reply:
I haven't modified the version of ipython notebook.